corehq/tests/nose.py

```diff
@@ -304,7 +340,12 @@ def teardown(self):
             return
 
         self.blob_db.close()
-        self.optimizer.__exit__(None, None, None)
+        if self.optimize:
+            self.optimizer.__exit__(None, None, None)
+
+        if os.environ.get("REUSE_DB") != "teardown":
```

Comment:
not blocking, but it could be nice to abstract out the interpretation of this environment variable.